src/covid19sim/simulator.py

```diff
@@ -942,10 +898,15 @@ def run(self, city):
             if (self.test_result != "positive" and
                 (self.test_recommended or
                 (self.is_incubated and self.env.timestamp - self.symptom_start_time >= datetime.timedelta(days=TEST_DAYS)))):
+                if (self.env.timestamp - self.test_results[0][1]).days == 0:
+                    warnings.warn(f"{self.name}'s last test time is less than a day. Will not retest human today. "
+                                  f"Current day {self.env.timestamp}, "
+                                  f"Last test time {self.test_results[0][1]}",
+                                  RuntimeWarning)
```

Comment:
Good catch! @satyaog Can you inform us whenever you find these bugs? It will help us too.